api/unpack/index.go

```diff
@@ -18,9 +19,13 @@ import (
 // MountPath is where this handler is supposed to be mounted
 const MountPath = "/nix/store/"
 
-var nixCache = libstore.HTTPBinaryCacheStore{
-	CacheURI: getEnv("NAR_CACHE_URI", "https://cache.nixos.org"),
-}
+var ctx = context.Background()
```

Comment:
Context objects are supposed to be passed around so they can be controlled. So setting one top-level here defeats its purpose.

Reply:
I agree, but I have to pass `ctx` to this function
```
var nixCache, err = libstore.NewBinaryCacheReader(ctx, "s3://nar?region=eu-west-1&endpoint=http://127.0.0.1:9000&scheme=http")
```

Do you have any suggestion?

Reply:
Did you try passing `context.Background()` directly to that function?

Reply:
That module design is not super clean as it was influenced by the restrictions from Vercel.

A better way would be to create a new struct that obeys the http.Handler interface, and initialize all of that is the top-level main function. That would make it more composable and more easily testable.

Reply:
pushed fix in 477d80549859a47897d4a2dceb6d9e13d1a1ce9b